src/basic/cgroup-util.c

```diff
@@ -132,13 +138,18 @@ int cg_read_pidref(FILE *f, PidRef *ret) {
         for (;;) {
                 pid_t pid;
 
-                r = cg_read_pid(f, &pid);
+                r = cg_read_pid(f, &pid, flags);
                 if (r < 0)
                         return r;
                 if (r == 0) {
                         *ret = PIDREF_NULL;
                         return 0;
                 }
+                if (pid == (pid_t) 0) {
+                        *ret = PIDREF_MAKE_FROM_PID(pid);
```

Comment:
This is very strange, as it is effectively `PIDREF_NULL`. And as per https://github.com/systemd/systemd/pull/32722#discussion_r1595238115, I think this should be an error.

How about renaming the flag to `CGROUP_DONT_SKIP_...`? Then, set the flag for `cg_read_pidref` in `unit_search_main_pid`.

Reply:
I've been thinking something similar just now. I could just have it straight up return -EREMOTE or -ESRCH here, instead of making this pseudo-NULL-pidref.

The flag currently basically is "CGROUP_DONT_SKIP_UNMAPPED", but negative names have been rejected previously, so it's called INCLUDE_UNMAPPED.

Reply:
`INCLUDE_UNMAPPED` is quite confusing when what you get is actually error. Also, there's `EXTRACT_DONT_COALESCE_SEPARATORS`. I don't think negative name in this case is awful.